src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -231,6 +231,12 @@ public async Task SaveAsync()
 
                 lock (this.lockObject)
                 {
+                    // If the tip is not the last item in the list skip saving.
+                    // This can happen if save is called in the middle of a reorg
+                    // then the pending collection can be temporary in an inconsistent state
+                    if (this.PendingBatch.Last().Key != this.pendingTipHashHeight.Height)
```

Comment:
Probably would be better to compare hashes instead of heights.

Reply:
But in general I'd actually do it a bit differently- we can proceed to saving items, just save all the items up to `pendingTipHashHeight`